src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/ExcusalRefusalLetterListRepositoryImpl.java

```diff
@@ -1,79 +1,193 @@
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.EXCUSE_POOL_MEMBER))
+                .and(QJurorHistory.jurorHistory.otherInformation.contains("Refuse Excuse"))
+                .and(QJurorHistory.jurorHistory.dateCreatedDateOnly.after(QJuror.juror.bureauTransferDate)))
+            .where(QJurorPool.jurorPool.isActive.isTrue()
+                .and(QJuror.juror.excusalRejected.eq("Y"))
+                .and(QJurorPool.jurorPool.owner.eq(owner)));
     }
 
+    private List<Tuple> getDatePrinted(String owner, List<String> poolNumbers) {
 
-    private void filterEligibleLetterSearchCriteria(JPAQuery<ExcusalRefusedLetterList> jpaQuery,
+        JPAQueryFactory queryFactory = new JPAQueryFactory(entityManager);
+
+        return queryFactory.select(
+                QPoolRequest.poolRequest.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJurorHistory.jurorHistory.dateCreated)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.eq(QJurorPool.jurorPool.juror))
+            .join(QPoolRequest.poolRequest).on(QPoolRequest.poolRequest.eq(QJurorPool.jurorPool.pool))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
+            .leftJoin(QJurorHistory.jurorHistory)
+            .on(QJurorHistory.jurorHistory.jurorNumber.eq(QJuror.juror.jurorNumber)
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.NON_EXCUSED_LETTER))
+                .and(QJurorHistory.jurorHistory.dateCreatedDateOnly.after(QJuror.juror.bureauTransferDate)))
+            .where(QJurorPool.jurorPool.isActive.isTrue()
+                .and(QJuror.juror.excusalRejected.eq("Y"))
+                .and(QJurorPool.jurorPool.owner.eq(owner))
+                .and(QPoolRequest.poolRequest.poolNumber.in(poolNumbers)))
+            .fetch();
+    }
+
+    private void filterEligibleLetterSearchCriteria(JPAQuery<Tuple> jpaQuery,
                                                     CourtLetterSearchCriteria courtLetterSearchCriteria) {
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.jurorNumber())) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.jurorNumber.startsWith(courtLetterSearchCriteria.jurorNumber()));
+            jpaQuery.where(QJurorPool.jurorPool.juror.jurorNumber.startsWith(courtLetterSearchCriteria.jurorNumber()));
         }
 
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.jurorName())) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.firstName.concat(" " + EXCUSAL_REFUSED_LETTER_LIST.lastName)
+            jpaQuery.where(QJurorPool.jurorPool.juror.firstName.concat(" ").concat(QJurorPool.jurorPool.juror.lastName)
```

Comment:
As you join on QJuror.juror we should filter on QJuror.juror to prevent duplicate joins

Reply:
updated